lib/pbench/test/unit/server/query_apis/test_datasets_delete.py

```diff
@@ -153,13 +153,12 @@ def test_query(
 
     def test_partial(
         self,
-        attach_dataset,
         caplog,
         client,
         get_document_map,
```

Comment:
Hmm; `get_document_map` is a non-SQL mock. Using `provide_metadata` with a real `server.index-map` metadata might work better?

Reply:
Nah! that didnt do the trick, I think because the detached instance is of the type `User`, going with a real metadata server index map didn't help.